### ISSUE TITLE
[feature request] DOS 3.3 FST, Pascal FST

### ISSUE THREAD

**Issue by ZoriaRPG** [a2stuff/a2d#200]
One thing that Apple II Desktop lacks, that GS/OS and GEOS managed to handle, is the ability to read DOS 3.3 format disks. I feel as if this could be solved, and that launching BASIC programmes from DOS 3.3 diskettes, and perhaps even BRUNning binary files from them should be possible in Apple II Desktop.

Perhaps it would be possible to backport GSOS FSTs, or just provide a driver for this. 

It would also be nice to access Pascal format volumes.

**Comment by inexorabletash** [comment]
The main limit is RAM - there's not enough room without substantial rework. But the rework is doable - more of the core of DeskTop could be moved into overlays.

The code is also very, very ProDOS-centric. Doable, but lots of work.

**Comment by ZoriaRPG** [comment]
Are you constraining future A2D releases to 64K, 128K, or expanded RAM? This is speculative, but IMO, most people who would regularly run A2D have expanded RAM and most likely at least a 3.5MHz system.

**Comment by inexorabletash** [comment]
128k is required now. A rearchitecture to support more than 128k if available would likely be a near total rewrite. Doable, but beyond the time investment I'm ever going to make. I'd encourage others to give it a try, though!

**Comment by ZoriaRPG** [comment]
So, basic DOS 3.3 support would exceed 128K. That's sad.

**Comment by inexorabletash** [comment]
Not necessarily, but memory is currently very tight - see the memory map in the desktop README. I don't know how big such FSTs would likely be, plus the abstraction layer on top of them replacing direct ProDOS MLI calls. I'm assuming several kilobytes, which means figuring out how to remove that much from DeskTop, either by optimization or overlays. 

GEOS is obviously a proof by example that it's doable in 128k, given that as a design goal from scratch. 

Contributions welcome.

**Comment by I3arana** [comment]
The two 8 bit packages that can read hfs disks are hfslink and a2fx. FYI
HFS link can read up to 40mb volumes.Both available on asimov
